src/merkle/tiered_smt/constants.rs

```diff
@@ -0,0 +1,400 @@
+use super::{Felt, RpoDigest, TieredSmt};
+
+/// Constants representing a path from an empty lower leaf to the root.
+///
+/// This array is indexed by the target depth, yielding the constant node value if that sub-tree is
+/// empty.
+pub const EMPTY_SUBTREES: [RpoDigest; 1 + TieredSmt::MAX_DEPTH as usize] = [
```

Comment:
That would probably require a crate split as the build script would have to depend on a portion of `miden-crypto`, generating a cyclic dependency.

It might be a good thing to split more complex data structures from crypto (even if inside a worskpace) as we might face this situation for other cases as well.

Reply:
I was thinking more along the route of making `Rpo256::merge()` be a `const fn` and then we can just compute all roots bottom up. But I'm not sure if having it as `const fn` is possible.

Reply:
We can have a dedicated method to do so, but we could face some issues as some operands of `Felt` are implemented as trait functions (besides `Hasher::merge` that we can easily circumvent by implementing an additional method).

This would probably mean we have to refactor `Felt` as `const fn` are not expected to be supported by the language anytime soon:

https://github.com/rust-lang/rfcs/blob/5c8d632e2a473764c7123524e6df17f97d7b3a07/text/0911-const-fn.md?plain=1#L83-L85